api/v1alpha1/validator/nicclusterpolicy_webhook.go

```diff
@@ -435,6 +479,159 @@ func validateRepository(repo string, allErrs field.ErrorList, fp *field.Path, ch
+			filepath.Join(manifestBaseDir, "state-ib-kubernetes"),
+		},
+		"nvIpam": {
+			w.Spec.NvIpam, state.NewStateNVIPAMCNI,
+			filepath.Join(manifestBaseDir, "state-nv-ipam-cni"),
+		},
+		"nicFeatureDiscovery": {
+			w.Spec.NicFeatureDiscovery, state.NewStateNICFeatureDiscovery,
+			filepath.Join(manifestBaseDir, "state-nic-feature-discovery"),
+		},
+	}
+	for stateName, renderData := range states {
+		localData := renderData
+		allErrs = validateContainerResourcesIfNotNil(&localData, policy, allErrs, fp, stateName)
+	}
+
+	if w.Spec.SecondaryNetwork != nil {
+		snfp := fp.Child("secondaryNetwork")
+
+		states := map[string]stateRenderData{
+			"cniPlugins": {
+				w.Spec.SecondaryNetwork.CniPlugins, state.NewStateCNIPlugins,
+				filepath.Join(manifestBaseDir, "state-container-networking-plugins"),
+			},
+			"ipoib": {
+				w.Spec.SecondaryNetwork.IPoIB, state.NewStateIPoIBCNI,
+				filepath.Join(manifestBaseDir, "state-ipoib-cni"),
+			},
+			"multus": {
+				w.Spec.SecondaryNetwork.Multus, state.NewStateMultusCNI,
+				filepath.Join(manifestBaseDir, "state-multus-cni"),
+			},
+			"ipamPlugin": {
+				w.Spec.SecondaryNetwork.IpamPlugin, state.NewStateWhereaboutsCNI,
+				filepath.Join(manifestBaseDir, "state-whereabouts-cni"),
+			},
+		}
+		for stateName, renderData := range states {
+			localData := renderData
+			allErrs = validateContainerResourcesIfNotNil(&localData, policy, allErrs, snfp, stateName)
+		}
+	}
+	return allErrs
+}
+
+func validateContainerResourcesIfNotNil(
+	resource *stateRenderData, policy *v1alpha1.NicClusterPolicy, allErrs field.ErrorList,
+	fp *field.Path, fieldName string) field.ErrorList {
+	if resource.spec == nil {
+		return allErrs
+	}
+
+	// Use reflection to check if ContainerResources is nil
+	val := reflect.ValueOf(resource.spec)
+	if val.Kind() == reflect.Ptr && !val.IsNil() {
+		imageSpec := val.Elem().FieldByName("ImageSpec")
+		if imageSpec.IsValid() {
+			containerResources := imageSpec.FieldByName("ContainerResources")
+			if containerResources.IsValid() && !containerResources.IsNil() {
+				_, renderer, err := resource.newState(nil, nil, resource.manifestDir)
+				if err != nil {
+					nicClusterPolicyLog.Error(err, "failed to created state renderer")
+				}
+				return validateResourceRequirements(
+					containerResources.Interface().([]v1alpha1.ResourceRequirements),
+					policy, renderer, allErrs, fp, fieldName)
+			}
+		}
+	}
+
+	return allErrs
+}
+
+func validateResourceRequirements(resources []v1alpha1.ResourceRequirements,
+	policy *v1alpha1.NicClusterPolicy, manifestRenderer state.ManifestRenderer,
+	allErrs field.ErrorList, fp *field.Path, child string) field.ErrorList {
+	if len(resources) == 0 {
+		allErrs = append(allErrs, field.Invalid(fp.Child(child).Child("containerResources"),
+			resources, "should not be empty if declared"))
+	}
+
+	supportedContainerNames, err := state.ParseContainerNames(manifestRenderer, policy, nicClusterPolicyLog)
```

Comment:
So we are rendering all the manifests on each webhook calls?
Isn't it a little heavy?